src/ncp/ncp_base.cpp

```diff
@@ -4493,6 +4566,75 @@ ThreadError NcpBase::SetPropertyHandler_STREAM_NET_INSECURE(uint8_t header, spin
     return errorCode;
 }
 
+#if OPENTHREAD_ENABLE_BORDER_AGENT_PROXY && OPENTHREAD_FTD
+ThreadError NcpBase::SetPropertyHandler_STREAM_BA_PROXY(uint8_t header, spinel_prop_key_t key, const uint8_t *value_ptr,
+                                                   uint16_t value_len)
+{
+    spinel_ssize_t parsedLength;
+    ThreadError errorCode = kThreadError_None;
+    const uint8_t *frame_ptr(NULL);
+    unsigned int frame_len(0);
+    uint16_t rloc;
+    uint16_t port;
+
+    // STREAM_BA_PROXY requires layer 2 security.
+    otMessage *message = otIp6NewMessage(mInstance, true);
+
+    if (message == NULL)
+    {
+        errorCode = kThreadError_NoBufs;
```

Comment:
Should this use one of the error handling macros instead of an if?
```
VerifyOrExit(message, errorCode = kThreadError_NoBufs);
...
...
exit:
return errorCode;
```